kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/security/auth/SimpleAclAuthorizer.java

```diff
@@ -166,8 +166,7 @@ public CompletableFuture<Boolean> canProduceAsync(KafkaPrincipal principal, Reso
 
     @Override
     public CompletableFuture<Boolean> canConsumeAsync(KafkaPrincipal principal, Resource resource) {
-        checkArgument(resource.getResourceType() == ResourceType.TOPIC,
-                String.format("Expected resource type is TOPIC, but have [%s]", resource.getResourceType()));
+        checkArgument(resource.getResourceType() == ResourceType.TOPIC);
```

Comment:
Can we extract a method like this: `private void checkArgument(ResourceType expected, ResourceType actual)`

Reply:
And we can print the error message in the `checkArgument` method when the type is unequal.